src/Pos/Explorer/Web/ClientTypes.hs

```diff
@@ -87,14 +87,15 @@ encodeHashHex :: Hash a -> Text
 encodeHashHex = decodeUtf8 . B16.encode . Bi.encodeStrict
 
 decodeHashHex :: Text -> Either Text (Hash a)
-decodeHashHex = fmap Bi.decode . processRes . B16.decode . encodeUtf8
-  where processRes (res, rest) =
-            if BS.null rest
-            then Right $ BSL.fromStrict res
-            else Left $ "decodeHashHex: couldn't decode rest of hash: " <> decodeUtf8 rest
-
-decodeHashHex' :: Text -> Hash a
-decodeHashHex' = either (error "decodeHashHex: invalid hash") identity . decodeHashHex
+decodeHashHex hashText = do
+    hashBinary <- processRes . B16.decode . encodeUtf8 $ hashText
```

Comment:
You can make your life easier if use `Serokell.Util.Base16.decode`:
https://github.com/serokell/serokell-util/blob/master/src/Serokell/Util/Base16.hs#L22